lib/OpenQA/Worker/Cache.pm

```diff
@@ -372,144 +285,114 @@ sub _update_asset_last_use {
 
 sub update_asset {
     my ($self, $asset, $etag, $size) = @_;
-
-    my $dbh = $self->dbh;
-
-    my $sql = "UPDATE assets set filename =?, etag =? , size = ?, last_use = strftime('%s','now') where filename = ?;";
-    my $res;
-
     eval {
-        $dbh->do('BEGIN EXCLUSIVE');
-        my $sth = $dbh->prepare($sql);
-        $sth->bind_param(1, $asset);
-        $sth->bind_param(2, $etag);
-        $sth->bind_param(3, $size);
-        $sth->bind_param(4, $asset);
-
-        $res = $sth->execute;
-        $dbh->commit;
+        my $tx  = $self->dbh->db->begin('exclusive');
+        my $sql = q(UPDATE assets set etag =? , size = ?, last_use = strftime('%s','now') where filename = ?;);
+        $self->dbh->db->query($sql, $etag, $size, $asset);
+        $tx->commit;
```

Comment:
Another case where you need to hold on to the db handle instead of calling `$self->dbh->db` repeatedly.